src/libs/hovercard/hovercard.js

```diff
@@ -0,0 +1,184 @@
+import { fetchProfile, fetchRelationship } from "../fetchPronouns.js";
+import { addShowMoreButton, generateProfile } from "./hovercard_generators.js";
+import { normaliseAccountName } from "../protootshelpers.js";
+import { hoverCardSettings } from "../settings.js";
+import { createElementWithClass } from "./hovercard_helpers.js";
+import { runtime } from "webextension-polyfill";
+
+const listenerTimeout = 200;
+
+let layer;
+let hovering = null;
+let hovercardExists = false;
+
+let mousePos = { x: undefined, y: undefined };
+
+/**
+ * Adds hovercard layer div to document body
+ */
+export function addHoverCardLayer() {
+	if (document.querySelector("#hovercardlayer")) return;
+
+	layer = createElementWithClass("div", "protoots-hovercardlayer");
+	layer.id = "hovercardlayer";
+
+	document.body.appendChild(layer);
+
+	window.addEventListener("mousemove", (event) => {
+		mousePos = { x: event.clientX, y: event.clientY };
+	});
+}
+
+/**
+ * Adds mouse enter and leave Eventlisteners to given element.
+ * Eventlisteners wait 200ms before adding or removing a hoverCard
+ * @param {HTMLElement} el Element to add Eventlisteners to
+ * @returns
+ */
+export function addHoverCardListener(el) {
+	if (el.nodeName != "ARTICLE") return;
+
+	const nametag = /** @type {HTMLElement|null} */ (el.querySelector(".display-name"));
+	const id = el.dataset.id;
+
+	nametag.addEventListener("mouseenter", () => {
+		hovering = nametag;
+		const enterpos = { x: mousePos.x, y: mousePos.y };
+		setTimeout(() => {
+			// if the cursor has moved more than 50 pixels since
+			// entering the name, don't generate a card
+			if (Math.hypot(mousePos.x - enterpos.x, mousePos.y - enterpos.y) > 50) return;
+			addHoverCard(nametag, id);
+		}, listenerTimeout);
+	});
+
+	nametag.addEventListener("mouseleave", () => {
+		hovering = null;
+		setTimeout(() => {
+			hovercardExists = false;
```

Comment:
what does hovercardExists mean?
if you hover again over the same nametag, it won't remove the hoverCard, but hovercardExists was already set to false. (no mental capacity right now to go over each possible combination of hovering and hoverCardExists)
(I'm not sure if grasp the intention right)